applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -75,60 +79,87 @@ public static Builder builder() {
      * The label value for the ConfigurationSetting is optional.
      *
      * @param setting The setting to add to the configuration service.
-     * @return ConfigurationSetting that was created or updated.
+     * @return ConfigurationSetting that was created.
      * @throws NullPointerException If {@code setting} is {@code null}.
-     * @throws IllegalArgumentException If {@link ConfigurationSetting#key()} is {@code null} or an empty string.
+     * @throws IllegalArgumentException If {@link ConfigurationSetting#key() key} or {@link ConfigurationSetting#value() value}
+     * are {@code null} or an empty string.
      * @throws ServiceRequestException If a ConfigurationSetting with the same key and label exists.
      */
     public Mono<Response<ConfigurationSetting>> addSetting(ConfigurationSetting setting) {
-        ConfigurationSetting result = validateSetting(setting);
-
+        ConfigurationSetting result = validateSetting(setting, true);
         return service.setKey(serviceEndpoint, result.key(), result.label(), result, null, getETagValue(ETAG_ANY));
```

Comment:
I get a little tripped up by the fact that addSetting and setSetting methods both call service.setKey. It's probably right, but it causes confusion when I'm skimming. Do you think a clarifying internal comment might be useful for future developers?

Reply:
Fixed. I added documentation around this.